### PR TITLE
Updated INT pre-2.0

### PR DESCRIPTION
**Pull request by orr101**
- Removed "Rep" 
- Removed special "Synthetic/Probe sections";  replaced with clarification of using C bit for such traffic 
- Allow Transit nodes to do OAM generation and NW Event Detection
- Allow Transit nodes to be INT Source for the Domain-specific parts 
- Some minor cleanup, length fixes in examples, etc.

### CODE REVIEW

**Comment by orr101** [comment]
INT_modes

**Comment by orr101** [comment]
Note need to also replace current 'INT modes" Drawing with this one

File: telemetry/specs/INT.mdk [hunk 1]
```diff
@@ -208,6 +208,7 @@ of Little Minions paper [^Minions].
 [^CLOVE]: CLOVE: Congestion-Aware Load Balancing at the Virtual Edge, ACM CoNEXT 2017
 [^Minions]: Millions of Little Minions: Using Packets for Low Latency Network Programming and Visibility, ACM SIGCOMM 2014.
 [^Postcard]: I Know What Your Packet Did Last Hop: Using Packet Histories to Troubleshoot Networks, USENIX NSDI 2014.
+[^Transit]: WHile this will be commonly done by Sink nodes, Transit nodes may also generate OAM's or carry out Network Event Detection
```
Comment (jklr) [inline telemetry/specs/INT.mdk:211]: WHile -> While

File: telemetry/specs/INT.mdk [hunk 2]
```diff
@@ -308,25 +309,27 @@ at the monitoring system to collate reports from multiple devices.
...
 
-* INT-CLONE-MD (eMbed Data): INT Source devices create extra traffic through
-cloning the original packets and then embedding the INT instructions and metadata into
-the clones only. INT Transit devices accumulate metadata only on the clones,
-INT Sink devices may send telemetry reports before discarding the clones.
-Compared to INT-MD, INT-CLONE-MD produces extra traffic
-and may not capture actual network state that the orignal packets experience, while
-this mode avoids direct modification on the original data packets.
-This mode is similar to IFA [^IFA].
+The only difference between live traffic and Synthetic traffic is that
```
Comment (jklr) [inline telemetry/specs/INT.mdk:318]: Synthetic -> synthetic

File: telemetry/specs/INT.mdk [hunk 3]
```diff
@@ -308,25 +309,27 @@ at the monitoring system to collate reports from multiple devices.
...
-This mode is similar to IFA [^IFA].
+The only difference between live traffic and Synthetic traffic is that 
+INT-Sink nodes may need to discard synthetic traffic after extracting 
+the collected INT data as opposed to forwading the traffic. This is 
+indicated by using the 'C' bit of the INT-Header to mark relavant packets
+as being Copies/Clones, to be discarded at the INT-Sink.  
 
-* INT-PROBE-MD (eMbed Data): similar to INT-CLONE-MD, but INT is applied to
-special probe packets instead of clone packets.
+All INT modes may be used on these Syntehtic/probe packets, as decided by the INT-Source node.
```
Comment (jklr) [inline telemetry/specs/INT.mdk:324]: Synthetic -> synthetic

File: telemetry/specs/INT.mdk [hunk 4]
```diff
@@ -308,25 +309,27 @@ at the monitoring system to collate reports from multiple devices.
...
+The only difference between live traffic and Synthetic traffic is that 
+INT-Sink nodes may need to discard synthetic traffic after extracting 
+the collected INT data as opposed to forwading the traffic. This is 
+indicated by using the 'C' bit of the INT-Header to mark relavant packets
+as being Copies/Clones, to be discarded at the INT-Sink.  
 
-* INT-PROBE-MD (eMbed Data): similar to INT-CLONE-MD, but INT is applied to
-special probe packets instead of clone packets.
+All INT modes may be used on these Syntehtic/probe packets, as decided by the INT-Source node. 
+Specifically the **INT-MD** (eMbed Data) mode applied to Synthetic or probe packets allows
```
Comment (jklr) [inline telemetry/specs/INT.mdk:325]: Synthetic -> synthetic

File: telemetry/specs/INT.mdk [hunk 5]
```diff
@@ -308,25 +309,27 @@ at the monitoring system to collate reports from multiple devices.
...
+indicated by using the 'C' bit of the INT-Header to mark relavant packets
+as being Copies/Clones, to be discarded at the INT-Sink.  
 
-* INT-PROBE-MD (eMbed Data): similar to INT-CLONE-MD, but INT is applied to
-special probe packets instead of clone packets.
+All INT modes may be used on these Syntehtic/probe packets, as decided by the INT-Source node. 
+Specifically the **INT-MD** (eMbed Data) mode applied to Synthetic or probe packets allows 
+functionality similar to **IFA**[^IFA] 
+
+It is likely that sytnetic traffic created by cloning would be discraded at the Sink,
```
Comment (jklr) [inline telemetry/specs/INT.mdk:328]: sytnetic -> synthetic

File: telemetry/specs/INT.mdk [hunk 6]
```diff
@@ -308,25 +309,27 @@ at the monitoring system to collate reports from multiple devices.
...
+while Probe packets might be marked for forwatrding or discarding, depending on the
```
Comment (jklr) [inline telemetry/specs/INT.mdk:329]: Probe -> probe
marked .. -> marked on the 'C' bit for discarding, or not marked for forwarding (something like this that clarifies the meaning of marking) 
forwatrding -> forwarding

File: telemetry/specs/INT.mdk [hunk 7]
```diff
@@ -912,38 +905,16 @@ INT Metadata Header and Metadata Stack:
...
-    - 3: Port-level and Next-hop-level replication requested.
-  - C (1b): Copy.
-    - If replication is requested for data packets, the INT Sink must be
-able to distinguish the original packet from replicas so that it can forward
-only original packets up the protocol stack, and drop all the replicas. The C
-bit must be set to 1 on each copy, whenever an INT hop replicates a packet.
-The original packet must have C bit set to 0.
-    - C bit must be set to 0 in the original packet by INT source
+  - Res (2b): Reserved
+  - C   (1b): Copy/Clone. INT Sink should Discard the packet after Extracting INT data
```
Comment (jklr) [inline telemetry/specs/INT.mdk:923]: Discard -> discard
Extracting -> extracting

File: telemetry/specs/INT.mdk [hunk 8]
```diff
@@ -1165,7 +1139,7 @@ INT Shim Header for TCP/UDP, INT type is hop-by-hop:
     0                   1                   2                   3
     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-   |     Type=1    |   Reserved    |  Length = 8   |   DSCP    |R R|
+   |     Type=1    |   Reserved    |  Length = 7   |   DSCP    |R R|
```
Comment (jklr) [inline telemetry/specs/INT.mdk:1151]: 7 -> 9 ?

(The shim header format is changed at the other PR: https://github.com/p4lang/p4-applications/pull/63 )

**Comment by jklr** [comment]
@rsivakolundu can you pls merge this PR? I will address the typos once merged into p4lang. Thx.

**Comment by rsivakolundu** [comment]
Merged.

-Ramesh

**Comment by jklr** [comment]
Thanks.

**Pull request merged**